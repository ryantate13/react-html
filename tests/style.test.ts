import {style} from '../src';

describe('style', () => {
    it('renders with children', () => {
        expect(style({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(style({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(style({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(style()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(style({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(style({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(style({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(style({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(style({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(style({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(style({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(style({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(style({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(style({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(style({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(style({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(style({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(style({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(style({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(style({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(style({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(style({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(style({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(style({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(style({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(style({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(style({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(style({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(style({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(style({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(style({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(style({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(style({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(style({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(style({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(style({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(style({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(style({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(style({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(style({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(style({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(style({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(style({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(style({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(style({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(style({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(style({translate: "no"})).toMatchSnapshot();
    });

    it('renders with media="screen"', () => {
        expect(style({media: "screen"})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(style({title: "TEST"})).toMatchSnapshot();
    });
});