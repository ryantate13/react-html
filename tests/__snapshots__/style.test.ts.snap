// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`style renders with accessKey="a,b,c,d" 1`] = `
<style
  accessKey="a,b,c,d"
/>
`;

exports[`style renders with autoCapitalize="characters" 1`] = `
<style
  autoCapitalize="characters"
/>
`;

exports[`style renders with autoCapitalize="none" 1`] = `
<style
  autoCapitalize="none"
/>
`;

exports[`style renders with autoCapitalize="off" 1`] = `
<style
  autoCapitalize="off"
/>
`;

exports[`style renders with autoCapitalize="on" 1`] = `
<style
  autoCapitalize="on"
/>
`;

exports[`style renders with autoCapitalize="sentences" 1`] = `
<style
  autoCapitalize="sentences"
/>
`;

exports[`style renders with autoCapitalize="words" 1`] = `
<style
  autoCapitalize="words"
/>
`;

exports[`style renders with children 1`] = `
<style>
  TEST
</style>
`;

exports[`style renders with className="foo bar baz" 1`] = `
<style
  className="foo bar baz"
/>
`;

exports[`style renders with contentEditable="false" 1`] = `
<style
  contentEditable="false"
/>
`;

exports[`style renders with contentEditable="true" 1`] = `
<style
  contentEditable="true"
/>
`;

exports[`style renders with dir="auto" 1`] = `
<style
  dir="auto"
/>
`;

exports[`style renders with dir="ltr" 1`] = `
<style
  dir="ltr"
/>
`;

exports[`style renders with dir="rtl" 1`] = `
<style
  dir="rtl"
/>
`;

exports[`style renders with draggable="false" 1`] = `
<style
  draggable="false"
/>
`;

exports[`style renders with draggable="true" 1`] = `
<style
  draggable="true"
/>
`;

exports[`style renders with empty attributes 1`] = `<style />`;

exports[`style renders with hidden="false" 1`] = `
<style
  hidden={false}
/>
`;

exports[`style renders with hidden="true" 1`] = `
<style
  hidden={true}
/>
`;

exports[`style renders with id="TEST" 1`] = `
<style
  id="TEST"
/>
`;

exports[`style renders with inputMode="decimal" 1`] = `
<style
  inputMode="decimal"
/>
`;

exports[`style renders with inputMode="email" 1`] = `
<style
  inputMode="email"
/>
`;

exports[`style renders with inputMode="none" 1`] = `
<style
  inputMode="none"
/>
`;

exports[`style renders with inputMode="numeric" 1`] = `
<style
  inputMode="numeric"
/>
`;

exports[`style renders with inputMode="search" 1`] = `
<style
  inputMode="search"
/>
`;

exports[`style renders with inputMode="tel" 1`] = `
<style
  inputMode="tel"
/>
`;

exports[`style renders with inputMode="text" 1`] = `
<style
  inputMode="text"
/>
`;

exports[`style renders with inputMode="url" 1`] = `
<style
  inputMode="url"
/>
`;

exports[`style renders with is="foo" 1`] = `
<style
  is="foo"
/>
`;

exports[`style renders with itemID="https://google.com/" 1`] = `
<style
  itemID="https://google.com/"
/>
`;

exports[`style renders with itemProp="https://google.com/" 1`] = `
<style
  itemProp="https://google.com/"
/>
`;

exports[`style renders with itemRef="foo bar" 1`] = `
<style
  itemRef="foo bar"
/>
`;

exports[`style renders with itemScope="false" 1`] = `
<style
  itemScope={false}
/>
`;

exports[`style renders with itemScope="true" 1`] = `
<style
  itemScope={true}
/>
`;

exports[`style renders with itemType="https://google.com/" 1`] = `
<style
  itemType="https://google.com/"
/>
`;

exports[`style renders with lang="" 1`] = `
<style
  lang=""
/>
`;

exports[`style renders with lang="es-419" 1`] = `
<style
  lang="es-419"
/>
`;

exports[`style renders with media="screen" 1`] = `
<style
  media="screen"
/>
`;

exports[`style renders with slot="TEST" 1`] = `
<style
  slot="TEST"
/>
`;

exports[`style renders with spellCheck="false" 1`] = `
<style
  spellCheck="false"
/>
`;

exports[`style renders with spellCheck="true" 1`] = `
<style
  spellCheck="true"
/>
`;

exports[`style renders with style="[object Object]" 1`] = `
<style
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`style renders with tabIndex="1" 1`] = `
<style
  tabIndex={1}
/>
`;

exports[`style renders with title="TEST" 1`] = `
<style
  title="TEST"
/>
`;

exports[`style renders with title="TEST" 2`] = `
<style
  title="TEST"
/>
`;

exports[`style renders with translate="no" 1`] = `
<style
  translate="no"
/>
`;

exports[`style renders with translate="yes" 1`] = `
<style
  translate="yes"
/>
`;

exports[`style renders without attributes 1`] = `<style />`;
